[ci] Internal builds broken - `npm install`

This broke with https://github.com/dotnet/aspire/pull/5212 which added a msbuild target to use `npm`.

Build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2512012&view=results

```
  'npm' is not recognized as an internal or external command,
  operable program or batch file.
D:\a\_work\1\s\playground\BrowserTelemetry\BrowserTelemetry.Web\BrowserTelemetry.Web.csproj(23,5): error MSB3073: The command "npm install" exited with code 9009.
##[error]playground\BrowserTelemetry\BrowserTelemetry.Web\BrowserTelemetry.Web.csproj(23,5): error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command "npm install" exited with code 9009.
```

cc @JamesNK 

<!-- Error message template  -->
## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

<!-- Use ErrorMessage for String.Contains matches. Use ErrorPattern for regex matches (single line/no backtracking). Set BuildRetry to `true` to retry builds with this error. Set ExcludeConsoleLog to `true` to skip helix logs analysis. -->

```json
{
  "ErrorMessage": "'npm' is not recognized as an internal or external command",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```
<!-- Known issue validation start -->
 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng/internal/_build/results?buildId=2512012
**Error message validated:** `['npm' is not recognized as an internal or external command`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 8/12/2024 6:39:38 PM UTC
<!-- Known issue validation end -->
<!--Known issue error report start -->

### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[2512012](https://dev.azure.com/dnceng/internal/_build/results?buildId=2512012)|dotnet-aspire|Build Repository|[Log](https://dev.azure.com/dnceng/7ea9116e-9fac-403d-b258-b31fcf1bb293/_apis/build/builds/2512012/logs/133)||
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|1|1|
<!--Known issue error report end -->

--- a/playground/BrowserTelemetry/BrowserTelemetry.Web/BrowserTelemetry.Web.csproj
+++ b/playground/BrowserTelemetry/BrowserTelemetry.Web/BrowserTelemetry.Web.csproj
@@ -4,13 +4,15 @@
     <TargetFramework>net8.0</TargetFramework>
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
+    <!-- Issue: https://github.com/dotnet/dnceng/issues/3091 -->
+    <NpmAvailable Condition="'$(ContinuousIntegrationBuild)' != 'true' and '$(OS)' != 'Windows_NT'">true</NpmAvailable>
   </PropertyGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\..\Playground.ServiceDefaults\Playground.ServiceDefaults.csproj" />
   </ItemGroup>
 
-  <Target Name="NpmInstall" Inputs="package.json" Outputs="node_modules/.install-stamp">
+  <Target Name="NpmInstall" Inputs="package.json" Outputs="node_modules/.install-stamp" Condition="'$(NpmAvailable)' == 'true'">
     <!--
         Use npm install or npm ci depending on RestorePackagesWithLockFile value.
         Uncomment the following lines if you want to use this feature:
@@ -30,7 +32,7 @@
       2. Run npm run build before building the .NET project.
       MSBuild runs NpmInstall before this task because of the DependsOnTargets attribute.
    -->
-  <Target Name="NpmRunBuild" DependsOnTargets="NpmInstall" BeforeTargets="BeforeBuild" Condition="'$(Configuration)' == 'Debug' ">
+  <Target Name="NpmRunBuild" DependsOnTargets="NpmInstall" BeforeTargets="BeforeBuild" Condition="'$(Configuration)' == 'Debug' and '$(NpmAvailable)' == 'true' ">
     <Exec Command="npm run build" />
   </Target>
 

--- a/playground/BrowserTelemetry/BrowserTelemetry.Web/BrowserTelemetry.Web.csproj
+++ b/playground/BrowserTelemetry/BrowserTelemetry.Web/BrowserTelemetry.Web.csproj
@@ -5,14 +5,14 @@
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
     <!-- Issue: https://github.com/dotnet/dnceng/issues/3091 -->
-    <NpmAvailable Condition="'$(ContinuousIntegrationBuild)' != 'true' and '$(OS)' != 'Windows_NT'">true</NpmAvailable>
+    <NpmUnavailable Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(OS)' == 'Windows_NT'">true</NpmUnavailable>
   </PropertyGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\..\Playground.ServiceDefaults\Playground.ServiceDefaults.csproj" />
   </ItemGroup>
 
-  <Target Name="NpmInstall" Inputs="package.json" Outputs="node_modules/.install-stamp" Condition="'$(NpmAvailable)' == 'true'">
+  <Target Name="NpmInstall" Inputs="package.json" Outputs="node_modules/.install-stamp" Condition="'$(NpmUnavailable)' != 'true'">
     <!--
         Use npm install or npm ci depending on RestorePackagesWithLockFile value.
         Uncomment the following lines if you want to use this feature:
@@ -32,7 +32,7 @@
       2. Run npm run build before building the .NET project.
       MSBuild runs NpmInstall before this task because of the DependsOnTargets attribute.
    -->
-  <Target Name="NpmRunBuild" DependsOnTargets="NpmInstall" BeforeTargets="BeforeBuild" Condition="'$(Configuration)' == 'Debug' and '$(NpmAvailable)' == 'true' ">
+  <Target Name="NpmRunBuild" DependsOnTargets="NpmInstall" BeforeTargets="BeforeBuild" Condition="'$(Configuration)' == 'Debug' and '$(NpmUnavailable)' != 'true' ">
     <Exec Command="npm run build" />
   </Target>
 

